_posts/2023-07-05-niklas-fuzzing.md

```diff
@@ -0,0 +1,59 @@
+---
+title: "Niklas Gögge on Fuzz Testing"
+permalink: /blog/2023/07/05/fuzzing/
+layout: post
+author: brink
+name: "brink"
+alt: Niklas Gögge on Fuzz Testing
+category: "News"
+description: Niklas Gögge on Fuzz Testing.
+---
+
+In a security-focused project like Bitcoin Core, testing is critical. [Bitcoin
+Core unit tests][] are written in C++ using the Boost libraries to quickly test
+components of the codebase in isolation (see the [bech32 unit tests][] for
+example). [Bitcoin Core functional tests][] are written in python to test
+features of the software, involving multiple components working together (see
+the [P2P ping functional tests][]).
+
+A less-familiar type of testing is fuzz testing where software repeatedly calls
+into portions of the Bitcoin Core codebase with random, potentially
+malformed or semi-valid inputs in order to surface potential negative consequences
+such as a crash, memory leak, or other failure scenarios.
```

Comment:
Could provide some examples of theoretical bugs that can be found, compelling but easy enough to understand.

"For example, fuzzing can uncover a series of specially-crafted p2p messages that leads to a crash, memory leak, or inconsistent internal state. As Bitcoin Core nodes are constantly exposed to anonymous peers on the network, a crash bug in the p2p message processing code can be catastrophic."

Reply:
Could also link to https://github.com/bitcoin-core/bitcoin-devwiki/wiki/Fuzz-Trophies, which is outdated now but still has a good list of old issues that were found.

Reply:
That’s great, added, thanks @dergoegge !